File: implementation/scipamato/scipamato-core-web/src/test/java/ch/difty/scipamato/core/web/paper/search/SearchOrderPanelInEditModeTest.java
package ch.difty.scipamato.core.web.paper.search;

import org.junit.jupiter.api.Test;

import ch.difty.scipamato.common.web.Mode;

class SearchOrderPanelInEditModeTest extends SearchOrderPanelTest {

    @Override
    Mode getMode() {
        return Mode.EDIT;
    }

    @Test
    void clickingDeleteIconLink() {
        getTester().startComponentInPage(makePanel());
        getTester().assertContains("foo");
        getTester().clickLink("panel:form:searchConditions:body:rows:1:cells:2:cell:link");
        getTester().assertInfoMessages("Removed foo");
        getTester().assertComponentOnAjaxResponse(PANEL_ID + ":form:searchConditions");
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/paper/search/SearchOrderPanelInEditModeTest.kt
package ch.difty.scipamato.core.web.paper.search

import ch.difty.scipamato.common.web.Mode
import org.junit.jupiter.api.Test

internal class SearchOrderPanelInEditModeTest : SearchOrderPanelTest() {

    override val mode: Mode
        get() = Mode.EDIT

    @Test
    fun clickingDeleteIconLink() {
        tester.startComponentInPage(makePanel())
        tester.assertContains("foo")
        tester.clickLink("panel:form:searchConditions:body:rows:1:cells:2:cell:link")
        tester.assertInfoMessages("Removed foo")
        tester.assertComponentOnAjaxResponse("$PANEL_ID:form:searchConditions")
    }
}
